app/src/main/java/org/wikipedia/gallery/GalleryActivity.kt

```diff
@@ -0,0 +1,761 @@
+        binding.licenseContainer.setOnLongClickListener { onLicenseLongClick() }
+    }
+
+    public override fun onDestroy() {
+        disposables.clear()
+        disposeImageCaptionDisposable()
+        binding.pager.unregisterOnPageChangeCallback(pageChangeListener!!)
+        pageChangeListener = null
+        TRANSITION_INFO = null
+        super.onDestroy()
+    }
+
+    public override fun onResume() {
+        super.onResume()
+        registerReceiver(downloadReceiver, IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE))
+        downloadReceiver.setCallback(downloadReceiverCallback)
+    }
+
+    public override fun onPause() {
+        super.onPause()
+        downloadReceiver.setCallback(null)
+        unregisterReceiver(downloadReceiver)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.menu_gallery, menu)
+        return true
+    }
+
+    override fun onDownload(item: GalleryItemFragment) {
+        item.imageTitle?.let {
+            funnel.logGallerySave(pageTitle, it.displayText)
+        }
+        if (item.imageTitle != null && item.mediaInfo != null) {
+            downloadReceiver.download(this, item.imageTitle!!, item.mediaInfo!!)
+            FeedbackUtil.showMessage(this, R.string.gallery_save_progress)
+        } else {
+            FeedbackUtil.showMessage(this, R.string.err_cannot_save_file)
+        }
+    }
+
+    override fun onShare(item: GalleryItemFragment, bitmap: Bitmap?, subject: String, title: PageTitle) {
+        item.imageTitle.let {
+            funnel.logGalleryShare(pageTitle, item.imageTitle!!.displayText)
+        }
+        if (bitmap != null && item.mediaInfo != null) {
+            ShareUtil.shareImage(this, bitmap,
+                File(ImageUrlUtil.getUrlForPreferredSize(item.mediaInfo!!.thumbUrl, Constants.PREFERRED_GALLERY_IMAGE_SIZE)).name,
+                subject, title.uri)
+        } else {
+            ShareUtil.shareText(this, title)
+        }
+    }
+
+    override fun setTheme() {
+        setTheme(Theme.DARK.resourceId)
+    }
+
+    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if ((requestCode == ACTIVITY_REQUEST_DESCRIPTION_EDIT || requestCode == ACTIVITY_REQUEST_ADD_IMAGE_TAGS) && resultCode == RESULT_OK) {
+            val action = if (data != null && data.hasExtra(Constants.INTENT_EXTRA_ACTION)) data.getSerializableExtra(
+                        Constants.INTENT_EXTRA_ACTION) as DescriptionEditActivity.Action?
+                        else if (requestCode == ACTIVITY_REQUEST_ADD_IMAGE_TAGS) DescriptionEditActivity.Action.ADD_IMAGE_TAGS else null
+            SuggestedEditsSnackbars.show(this, action, true, targetLanguageCode,
+                action === DescriptionEditActivity.Action.ADD_IMAGE_TAGS,
+                object : SuggestedEditsSnackbars.OpenPageListener {
```

Comment:
We can add `fun` to the ` interface OpenPageListener ` in order to use the lambda expression here.